src/open_clip/hf_model.py

```diff
@@ -3,45 +3,91 @@
 Wraps HuggingFace transformers (https://github.com/huggingface/transformers) models for use as a text tower in CLIP model.
 """
 
+import re
+
 import torch
 import torch.nn as nn
 from torch import TensorType
-from traitlets import default
 try:
     import transformers
     from transformers import AutoModel, AutoTokenizer, AutoConfig, PretrainedConfig
+    from transformers.modeling_outputs import BaseModelOutput, BaseModelOutputWithPooling
 except ImportError as e:
     transformers = None
 
 from timm.models.layers import Mlp
 
-
 # utils
+def _camel2snake(s):
+    return re.sub(r'(?<!^)(?=[A-Z])', '_', s).lower()
+
 # TODO: cls, max, mean, last
 _POOLERS = {}
 
 def register_pooler(cls):
-    "Register pooler class"
-    pass
+    "Decorator registering pooler class"
+    _POOLERS[_camel2snake(cls.__name__)] = cls
+    return cls
 
+@register_pooler
 class DummyPooler(nn.Module):
     "Fetches first of output hidden state"
 
     def __init__(self):
         super().__init__()
         pass
     
-    def forward(self, x):
+    def forward(self, x:BaseModelOutput, attention_mask:TensorType):
         return x.last_hidden_state[:, 0, :]
 
+@register_pooler
+class MeanPooler(nn.Module):
+    "Mean pooling"
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def forward(self, x:BaseModelOutput, attention_mask:TensorType):
+        masked_output = x.last_hidden_state * attention_mask.unsqueeze(-1)        
+        return masked_output.sum(dim=1) / attention_mask.sum(-1, keepdim=True)
+
+@register_pooler
+class MaxPooler(nn.Module):
```

Comment:
missing '__init__'. If not needed here then why does MeanPooler and DummyPooler have one?

Reply:
good point, removed `__ini__` from `MeanPooler` for consistency, I've also removed `DummyPooler` - don't think we'll need it